src/Chapter3.hs

```diff
@@ -1006,6 +1128,99 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+class Fighter a where
+  attack :: a -> Int
+  health  :: a -> Int
+  takeTurn :: a -> Fightable -> Fight
+  receiveDamage :: a -> Int -> a
+
+data Fightable =
+    FightHuman Human
+  | FightBeast Beast
+    deriving Show
```

Comment:
Maybe not. I did it because I wanted to be able to have a `Fight` data constructor have two fields which were the same type but could be Beast or Fighter

Reply:
Thanks for clarifications!
The typeclasses are useful for such cases, so you can use the same function on different types if they have necessary instances 🙂 

Let me know if we could discuss this task and typeclasses more, would be happy to chat! 👍🏼